### PR TITLE
new versions of `TFEL` and `MGIS`

### PR DESCRIPTION
**Pull request by thelfer**

### CODE REVIEW

File: var/spack/repos/builtin/packages/tfel/package.py [hunk 1]
```diff
@@ -81,6 +86,12 @@ class Tfel(CMakePackage):
     variant('java', default=False,
             description='Enables java interface')
 
+    # only since TFEL-3.3
+    variant('comsol', default=False,
+            description='Enables comsol interface')
+    variant('diana-fea', default=False,
+            description='Enables comsol interface')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/tfel/package.py]: Do these variants need to be used in `cmake_args` below?

Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: Yes, they do. I have seen an Issue about specifying compatible versions in `variant`

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tfel/package.py]: Nope, the best you can do is use a `conflicts` statement

Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: @adamjstewart Any example ?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tfel/package.py]: You can do something like:
```python
conflicts('+comsol', when='@:3.2', msg='comsol interface only supported in TFEL-3.3+')
```
Also, it looks like you need to fix the description for `diana-fea`.

In the `cmake_args` function, you can do:
```python
if self.spec.satisfies('@3.3:'):
    if '+comsol' in self.spec:
        args.append('-DENABLE_COMSOL=ON')
    else:
        args.append('-DENABLE_COMSOL=OFF')
```
This is just an example, I don't know what the appropriate CMake flags are.

**Comment by thelfer** [comment]
@adamjstewart Thanks for the review. I'll modify this ASAP.

File: var/spack/repos/builtin/packages/tfel/package.py (hunk 1 above)
Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: > You can do something like:
> 
> ```python
> conflicts('+comsol', when='@:3.2', msg='comsol interface only supported in TFEL-3.3+')
> ```

Well, I have questions about this syntax:

- would it work with the rliv-3.3 and the master branches  ?
- does it mean that it won't work with 3.2.1 ?

> Also, it looks like you need to fix the description for `diana-fea`.

Done ! Thanks.

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tfel/package.py]: The syntax `@:3.2` means all versions less than or equal to `3.2.X`, so 3.2.1 would also raise a conflict unless you say `@:3.2.0`. Spack sorts versions like `master` and `develop` higher than all numeric releases, so the variant would work for the `master` branch. Unfortunately, all other non-numeric versions like `rliv-3.3` would be sorted last. I think you can do:
```python
conflicts('+comsol', when='@:rliv-3.2,0:3.2', msg='...')
```
and:
```python
if self.spec.satisfies('@rliv-3.3:rliv-999,3.3:'):
```
You can test this and play around with it like so:
```python
$ spack python
Spack version 0.13.3
Python 3.7.4, Darwin x86_64
>>> from spack.version import ver
>>> ver('3.2.1') in ver(':3.2')
True
>>> ver('3.2.1') in ver(':3.2.0')
False
>>> ver('rliv-3.2') in ver(':rliv-3.2,0:3.2')
True
>>> ver('rliv-3.3') in ver(':rliv-3.2,0:3.2')
False
```

Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: @adamjstewart  That's very nice !
I'll try to update my package following your suggestion.
However, this is not blocking for this PR, does it ?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/tfel/package.py]: You're the maintainer, you get to decide 😄 

Do you want me to merge things as is and you can hack on this in a follow-up PR?

Comment (thelfer) [reply var/spack/repos/builtin/packages/tfel/package.py]: Yep. That's exactly what I have in mind. Thanks a lot.

**Comment by adamjstewart** [comment]
The only thing preventing merging is this flake8 error:
```
var/spack/repos/builtin/packages/tfel/package.py:145: [W293] blank line contains whitespace
```

**Pull request merged**